### PR TITLE
Move `after_pipeline|training_evaluation_workers` to pipeline config

### PR DESCRIPTION
**Pull request by XianzheMa**
Per the title, this solves https://github.com/eth-easl/modyn/issues/508.

### CODE REVIEW

File: modyn/supervisor/internal/pipeline_executor/evaluation_executor.py [hunk 1]
```diff
@@ -156,7 +156,9 @@ def run_pipeline_evaluations(
             )
             eval_requests += handler_eval_requests
 
-        num_workers = self.config.supervisor.after_training_evaluation_workers if self.config.supervisor else 1
+        if len(eval_requests) == 0:
+            return SupervisorLogs()
+        num_workers = self.pipeline.evaluation.after_training_evaluation_workers
```
Comment (XianzheMa) [inline modyn/supervisor/internal/pipeline_executor/evaluation_executor.py:161]: here we don't need to assert on self.pipeline.evaluation, as it is asserted in the beginning of this func.

**Review by MaxiBoether — APPROVED** [review]
Thank you!

File: codecov.yml [hunk 2]
```diff
@@ -3,3 +3,5 @@ coverage:
     project:
       default:
         informational: true
+    patch:
+      informational: true
```
Comment (XianzheMa) [inline codecov.yml:7]: We somehow don't have a unit test file for the quite large `evaluation_executor.py`.  https://github.com/eth-easl/modyn/pull/490

@robinholzi Is there a reason? If it is no intentional please add one!

I will for the moment also make `patch` action informational. As it does not make sense for me to create a huge unit test only to cover these trivial changes.

**Comment by XianzheMa** [comment]
I need to resolve the comment to merge the PR, so I copy-paste my question here:
We somehow don't have a unit test file for the quite large evaluation_executor.py. https://github.com/eth-easl/modyn/pull/490

@robinholzi  Is there a reason? If the missing unit test is not intentional please add one!

I will for the moment also make patch action informational. As it does not make sense for me to create a huge unit test only to cover these trivial changes.

**Comment by robinholzi** [comment]
@XianzheMa I guess that depends on what you define as a `unit`. We are testing the logic in `evaluation_executor` with the logic from `test_pipeline_supervisor` just like we did before the introduction of `evaluation_executor`. As there are some uncovered parts though, I will add dedicated tests for them! Thanks for the pointer!

**Pull request merged**